js/apps/admin-ui/src/components/time-selector/TimeSelector.tsx

```diff
@@ -35,17 +35,17 @@ export type TimeSelectorProps = Omit<
     className?: string;
   };
 
-export const getTimeUnit = (value: number | undefined = 0) =>
-  allTimes.reduce(
+const getTimeUnit = (units: TimeUnit[], value: number | undefined = 0) =>
+  units.reduce(
     (v, time) =>
       value % time.multiplier === 0 && v.multiplier < time.multiplier
         ? time
         : v,
-    allTimes[0],
-  );
+    units[0],
```

Comment:
Are we sure this is correct? It looks like the reduce starts with `units[0]`, but then on the first entry of the reduce `v` will also be the same value as `units[0]`. It also seems that as long the modulo `multiplier` is reached `v` is returned, so perhaps this could be accomplished with `for` loop and `return` as well? Avoiding additional iteration the proccess.

Reply:
`units[0]` is the default it only iterates once, you could do it with a for loop of course:
```js
let largestUnit = units[0];
for (let i = 1; i < units.length; i++) {
  const unit = units[i];
  if (value % unit.multiplier === 0 && largestUnit.multiplier < unit.multiplier) {
    largestUnit = unit;
  }
}
```
But this is just a lot longer

Reply:
Ahh yeah, the default makes sense